functions/go/set-image/README.md

```diff
@@ -0,0 +1,118 @@
+# set-image
+
+## Overview
+
+<!--mdtogo:Short-->
+
+The `set-image` function sets the image for all instances of a provided image
+`name` with the provided image `newName:newTag` or `newName@digest`. A common
+deployment pattern is to update images to the latest tags or, ideally, digests.
+Sometimes people want to change the image repo, also, such as for dev builds.
+
+<!--mdtogo-->
+
+You can learn more about images [here][image].
+
+<!--mdtogo:Long-->
+
+## Usage
+
+This function can be used with any KRM function orchestrators (e.g. kpt).
+
+For all resources which specify an image of a given `name`, the `set-image`
+function sets that image field to the specified `newName:newTag` or
+`newName@digest`.
+
+By default the function updates the standard [fields][commonimage] used to
+specify an image.
+
+This function can be used both declaratively and imperatively.
+
+### FunctionConfig
+
+There are 2 kinds of `functionConfig` supported by this function:
+
+- `ConfigMap`
+- A custom resource of kind `SetImage`
+
+To use a `ConfigMap` as the `functionConfig`, the desired image spec must be
+specified using the following fields:
+- `data.name`
+- `data.newName`
+- `data.newTag` or `data.digest`
+
+To set the image `nginx` to `bitnami/nginx:1.21.4` for all resources, we use the
+following `functionConfig`:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  name: nginx
+  newName: bitnami/nginx
+  newTag: 1.21.4
+```
+
+To set an image `nginx` to
+`nginx@sha256:3cbbd11b65aab276c8578c039d0c21d0ffb7a496e09c0f632bac1a1b2c115256`
+for all resources, we use the following `functionConfig`:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  name: nginx
+  newName: nginx
+  digest: sha256:3cbbd11b65aab276c8578c039d0c21d0ffb7a496e09c0f632bac1a1b2c115256
+```
+
+To use a `SetImage` custom resource as the `functionConfig`, the desired
```

Comment:
Per https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/_template/README.md#functionconfig, We should describe the fields more precisely.

Reply:
I've updated some of the descriptions here - let me know if you find them to be more precise now. Note that I based this doc off of similar docs from the other functions, e.g.
https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/set-annotations/README.md
https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/set-namespace/README.md
etc.